packages/webapp/src/components/ImagePicker/index.tsx

```diff
@@ -44,16 +44,19 @@ type CommonProps = {
   optional?: boolean;
   defaultUrl?: string;
   isDisabled?: boolean;
+  shouldReset?: boolean;
 };
 
 type CustomFileUpload = CommonProps & {
   onSelectImage?: never;
   onFileUpload: OnFileUpload;
+  shouldReset?: never;
 };
 
 type DirectImageUpload = CommonProps & {
   onSelectImage: (file: File) => void;
   onFileUpload?: never;
+  shouldReset?: boolean;
```

Comment:
@SayakaOno from what you remember of working in ImagePicker, does this look right? In my understanding of DirectImageUpload, the previewUrl is never controlled, just the file blob itself? So if the parent form needs to reset the preview from the outside (on form clear) it needs something like this?

Reply:
Sorry I needed some time to think this through...

If we had more time, I think it would be better to refactor the component to either:
1. Accept a `ref` and clear the file from the parent, similar to how it’s done for ReactSelect here:
     ![Screenshot 2024-11-26 at 1 19 15 PM](https://github.com/user-attachments/assets/67881c90-7adc-4d0e-a2fb-46ec8596f30e)
2. Take file and/or previewUrl as props

I don’t think the hook can handle image removal without these refactors.
That said, refreshing the form seems like the best approach in this case!